### PR TITLE
Fixes #12470 - skip post-sync actions if no new content was synced

### PR DESCRIPTION
**Pull request by jlsherrill**
also adds setting to force those actions

### CODE REVIEW

File: app/lib/actions/pulp/repository/sync.rb [hunk 1]
```diff
@@ -27,11 +27,26 @@ def invoke_external_task
...
             pulp_tasks
           end
         end
 
+        def external_task=(tasks)
+          output[:contents_changed] = content_changed?(tasks)
+          super
+        end
+
+        def content_changed?(tasks)
```
Comment (daviddavis) [inline app/lib/actions/pulp/repository/sync.rb]: Some places you use contents_changed and others you use content_changed. Might be nice to use the same terminology.

File: app/lib/actions/katello/repository/errata_mail.rb [hunk 2]
```diff
@@ -2,9 +2,9 @@ module Actions
   module Katello
     module Repository
       class ErrataMail < Actions::EntryAction
-        def plan(repo, last_updated = nil)
+        def plan(repo, last_updated = nil, contents_changed = nil)
```
Comment (daviddavis) [inline app/lib/actions/katello/repository/errata_mail.rb]: Would `false` be a better default here for contents_changed?

File: app/models/setting/katello.rb [hunk 3]
```diff
@@ -14,7 +14,8 @@ def self.load_defaults
         self.set('content_action_finish_timeout', N_("Time in seconds to wait for a Host to finish a remote action"), 3600),
         self.set('pulp_sync_node_action_accept_timeout', N_("Time in seconds to wait for a pulp node to remote action"), 20),
         self.set('pulp_sync_node_action_finish_timeout', N_("Time in seconds to wait for a pulp node to finish sync"), 12.hours.to_i),
-        self.set('check_services_before_actions', N_("Whether or not to check the status of backend services such as pulp and candlepin prior to performing some actions."), true)
+        self.set('check_services_before_actions', N_("Whether or not to check the status of backend services such as pulp and candlepin prior to performing some actions."), true),
+        self.set('force_post_sync_actions', N_("Force post sync actions such as indexing and email even if no content was available."), false)
```
Comment (daviddavis) [inline app/models/setting/katello.rb]: Might be nice to state what kind of emails here. The term of just email had me a bit baffled until I remember I think we send out errata applicability emails.

**Comment by daviddavis** [comment]
Some small comments. Also, I am not sure about the name SyncContentsChanged. This is a sort of check to see whether content changed? What about something like CheckContentsChanged or something? Or is this middleware suppose to contain more logic at some point?

File: app/lib/actions/katello/repository/errata_mail.rb (hunk 2 above)
Comment (daviddavis) [inline app/lib/actions/katello/repository/errata_mail.rb]: By the way, where is contents_changed being used in ErrataMail?

Comment (jlsherrill) [inline app/lib/actions/katello/repository/errata_mail.rb]: good catch, was missing the middleware

Comment (jlsherrill) [inline app/lib/actions/katello/repository/errata_mail.rb]: was also missing from RegenerateApplicability

**Comment by daviddavis** [comment]
APJ

**Pull request merged**